### PR TITLE
Feature

### PR DESCRIPTION
**Pull request by AdfPizarro**
Fixed the issues and created another pull request as requested by [https://github.com/maelfosso](url)
[https://github.com/gdumani/Ruby_enumerables/pull/4](url)

### CODE REVIEW

**Comment by AdfPizarro** [comment]
Disabled cyclomatic complexity and PerceivedComplexity in order to make the function work for RegExp, Class Names, variables, Arrays and nill values

File: enumerables.rb
Comment (shshamim63) [inline enumerables.rb]: multiply_els is not an enumerable method to it should be placed outside of the module. It accepts an array as an argument and multiplies all the elements of the array together using #my_inject. A method accepts an array as an argument. So the final outcome of this function should look like...
```ruby
multiply_els(arr)
  arr.my_inject(3, :*)
end
```

File: enumerables.rb
Comment (shshamim63) [inline enumerables.rb]: my_map` identical and returns the same thing as ruby's [`map`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-map). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_map.png) from [Odin](https://github.com/TheOdinProject/curriculum/blob/master/ruby_programming/archive/basic_ruby/project_advanced_building_blocks.md#project-2-enumerable-methods))
- executes only the proc when both a block and a proc are given and returns a new array with the results of running the proc once for every element in enum.
```ruby
my_proc = Proc.new { |num| num > 10 }
[11, 2, 3, 15].my_map(my_proc) {|num| num < 10 } # should return [true, false, false, true]
```
Based on your condition you are checking if block_given? and it is finding block all the time based on the example that I have given. So you should add another clause that is and !oper. So the condition will become
```ruby
if block_given? and !oper
```

File: enumerables.rb
Comment (shshamim63) [inline enumerables.rb]: Please remove the commented code.

File: enumerables.rb
Comment (shshamim63) [inline enumerables.rb:100]: `my_inject` identical and returns the same thing as ruby's [`inject`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-inject). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_inject.png) from [Odin](https://github.com/TheOdinProject/curriculum/blob/master/ruby_programming/archive/basic_ruby/project_advanced_building_blocks.md#project-2-enumerable-methods))
- raises a "LocalJumpError" when no block or argument is given. But the current configuration does not cover it.

File: enumerables.rb
Comment (shshamim63) [inline enumerables.rb]: `my_none?` identical and returns the same thing as ruby's [`none?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-none-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_none.png) from [Odin](https://github.com/TheOdinProject/curriculum/blob/master/ruby_programming/archive/basic_ruby/project_advanced_building_blocks.md#project-2-enumerable-methods))
- returns true if the block never returns true for all elements
```ruby
print [1, 2, 3].my_none? {|i| i > 4} # should print true but your one throws an error.
```
- returns false if the block ever returns true for all elements.
```ruby
print [1, 2, 3].my_none? {|i| i < 4} # should print false but your one throws an error.
```
- returns false if the block ever returns true for all elements::range
```ruby
print (1...3).my_none? {|i| i < 4} # should print false but your one throws an error.
```

**Review by shshamim63 — CHANGES_REQUESTED** [review]
## Status :dart: Changes Required
___
Hi @AdfPizarro and @gdumani :wave: 
Amazing job. Almost near to complete this project. I have found some simple issues and you guys can easily resolve them. Let's check them out.
___
### :spades: Mandatory Changes :spades: 
___
- In the readme, the maintainer list are breaking their flow.
image
- Check out the comments attached to their respective block comment and resolve them.
  - my_none?
  - my_inject
  - my_map
  - multiply_els
___
##### Please fix the above-mentioned issue and send another review request.
___
## :herb: Happy Coding :herb: 
### Shakhawat Hossain

File: enumerables.rb
Comment (AdfPizarro) [reply enumerables.rb]: Removed the code

File: enumerables.rb
Comment (AdfPizarro) [reply enumerables.rb]: moved the multiply outside the enumerables

File: enumerables.rb
Comment (AdfPizarro) [reply enumerables.rb]: Thanks we updated it

File: enumerables.rb
Comment (AdfPizarro) [reply enumerables.rb:100]: Added the raise local jump error

File: enumerables.rb
Comment (AdfPizarro) [reply enumerables.rb]: Updated method

**Review by kobiyoyo — CHANGES_REQUESTED** [review]
## Hi Team      
#### Great Job so far, but there are a few more things required. 
### General
- [x] `my_none?`:
- When no block or argument is given returns true only if none of the collection members is true
#### for example
`p  [nil, false, true, []].none?`   return `false`
- When a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern
#### for example
`p ['hello'].none?(5)` -  return `true`
- when a class is passed as an argument returns true if none of the collection is a member of such class
#### for example
`p [3,3,3,3].none?(String)`  return `true`
- [x] `my_inject`:
- When a symbol is specified combines each element of the collection by applying the symbol as a named method
#### for example
`p (5..10).inject(2, :*) -  return 302400`
### Make changes and submit another code review request.
*Happy coding* 
(Any questions Feel free to contact me on slack @kobiyoyo)

**Review by rOluochKe — APPROVED** [review]
Hello Team
### Good job and for making the changes :clap: Your project has been approved. Congratulations :boom: :100:
### Merge and submit. Good luck with the next project. 
- Happy coding!

**Comment by AdfPizarro** [comment]
Note for the reviewer, the project was already approved but my partner has the project as a solo one and I have this as a cooperative because he started one month before but left the program for one month, so was already approved but I can not submit the completed form because of that

**Review by mosesogwo — COMMENTED** [review]
**Hi @AdfPizarro, you should be able to submit the project completion form now. :heavy_check_mark:**

**Pull request merged**
